joint_trajectory_controller/CMakeLists.txt

```diff
@@ -42,6 +42,11 @@ catkin_package(
 
 include_directories(include ${Boost_INCLUDE_DIR} ${catkin_INCLUDE_DIRS})
 
+if(CATKIN_ENABLE_TESTING AND ENABLE_COVERAGE_TESTING)
+  find_package(code_coverage REQUIRED)
+  APPEND_COVERAGE_COMPILER_FLAGS()
+endif()
```

Comment:
Any reason we have two separate `if`s? I think we should move the `find_package` and `APPEND_COVERAGE_COMPILER_FLAGS` into the other `if(ENABLE_COVERAGE_TESTING)` below.

Reply:
I think the idea is to append the compiler flags before defining any targets. See the usage example in https://github.com/mikeferguson/code_coverage

Reply:
Ah, makes sense